File: recycler-adapters/src/main/java/com/dyhdyh/adapters/holder/BaseViewHolder.java
package com.dyhdyh.adapters.holder;

import android.support.annotation.LayoutRes;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * @author dengyuhan
 * created 2018/5/7 16:05
 */
public class BaseViewHolder<T> extends RecyclerView.ViewHolder {

    public BaseViewHolder(@LayoutRes int layoutId, ViewGroup parent) {
        this(LayoutInflater.from(parent.getContext()).inflate(layoutId, parent, false));
    }

    public BaseViewHolder(View itemView) {
        super(itemView);
    }

    public void bindViewHolder(T item, int position) {

    }

}

File: recycler-adapters/src/main/java/in/xiandan/adapters/BaseViewHolder.kt
package `in`.xiandan.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by xiandanin on 2021/1/12 14:43
 */
open class BaseViewHolder : RecyclerView.ViewHolder {
    constructor(parent: ViewGroup, @LayoutRes layoutId: Int) : super(LayoutInflater.from(parent.context).inflate(layoutId, parent, false))

    constructor(itemView: View) : super(itemView)

}